### PR TITLE
Add release numbering.

### PR DESCRIPTION
**Pull request by CyrusRoshan**

### CODE REVIEW

File: pkg/controller/service/service.go [hunk 1]
```diff
@@ -2134,7 +2135,22 @@ func (s *Service) getRelease(w http.ResponseWriter, r *http.Request,
 	vars := mux.Vars(r)
 	releaseID := vars["release"]
 
-	release, err := s.releases.GetRelease(r.Context(), releaseID, projectID, applicationID)
+	var release *models.Release
+	var err error
+
+	if strings.Contains(releaseID, "_") {
```
Comment (joshwget) [inline pkg/controller/service/service.go]: Can we move this to a middleware function (`withRelease`)?

File: pkg/controller/store/mysql/schema.sql [hunk 2]
```diff
@@ -374,6 +375,7 @@ create table if not exists releases (
   created_by_service_account_id varchar(32),
 
   primary key (id),
+  unique application_release_count (`number`, application_id),
```
Comment (joshwget) [inline pkg/controller/store/mysql/schema.sql]: Even though it really doesn't matter to much, can we make this `(`number`, application_id, project_id)` for consistency?

File: pkg/controller/store/mysql/sql.go [hunk 3]
```diff
@@ -663,25 +663,32 @@ const getApplicationDeviceCounts = `
...
 `
 
 // Index: project_id_application_id_id
 const getRelease = `
-  select id, created_at, project_id, application_id, config, raw_config, created_by_user_id, created_by_service_account_id from releases
+  select id, ` + "`number`" + `, created_at, project_id, application_id, config, raw_config, created_by_user_id, created_by_service_account_id from releases
   where id = ? and project_id = ? and application_id = ?
 `
 
+// Index: project_id_application_id_id
```
Comment (joshwget) [inline pkg/controller/store/mysql/sql.go]: We'll need a new index for this

File: pkg/controller/service/service.go (hunk 1 above)
Comment (CyrusRoshan) [reply pkg/controller/service/service.go]: Considered it, but we're only using it for one endpoint for now

Comment (CyrusRoshan) [reply pkg/controller/service/service.go]: I guess we might use it in the future

File: pkg/controller/store/mysql/schema.sql (hunk 2 above)
Comment (CyrusRoshan) [reply pkg/controller/store/mysql/schema.sql]: Sure

File: pkg/controller/service/service.go (hunk 1 above)
Comment (CyrusRoshan) [reply pkg/controller/service/service.go]: Want to pass in the release by pointer or by ID?

Comment (joshwget) [reply pkg/controller/service/service.go]: Object, no pointer, is probably best. It should never be nil.

Also fair to wait until later to do this.

Comment (CyrusRoshan) [reply pkg/controller/service/service.go]: Done, I'll refactor the rest later

**Pull request merged**